pkg/accounts/adaptor/controller/account.ts

```diff
@@ -0,0 +1,283 @@
+import { type UnfollowService } from '../../service/unfollow.js';
+import {
+  type CreateAccountResponseSchema,
+  type GetAccountResponseSchema,
+  type LoginResponseSchema,
+  type UpdateAccountResponseSchema,
+} from '../validator/schema.js';
+
+export class AccountController {
+  private readonly registerAccountService: RegisterAccountService;
+  private readonly editAccountService: EditAccountService;
+  private readonly fetchAccountService: FetchAccountService;
+  private readonly freezeService: FreezeService;
+  private readonly tokenVerifyService: TokenVerifyService;
+  private readonly authenticationService: AuthenticationService;
+  private readonly silenceService: SilenceService;
+  private readonly followService: FollowService;
+  private readonly unFollowService: UnfollowService;
+  private readonly resendTokenService: ResendVerifyTokenService;
+
+  constructor(args: {
+    registerAccountService: RegisterAccountService;
+    editAccountService: EditAccountService;
+    fetchAccountService: FetchAccountService;
+    freezeService: FreezeService;
+    tokenVerifyService: TokenVerifyService;
+    authenticationService: AuthenticationService;
+    silenceService: SilenceService;
+    followService: FollowService;
+    unFollowService: UnfollowService;
+    resendTokenService: ResendVerifyTokenService;
+  }) {
+    this.registerAccountService = args.registerAccountService;
+    this.editAccountService = args.editAccountService;
+    this.fetchAccountService = args.fetchAccountService;
+    this.freezeService = args.freezeService;
+    this.tokenVerifyService = args.tokenVerifyService;
+    this.authenticationService = args.authenticationService;
+    this.silenceService = args.silenceService;
+    this.followService = args.followService;
+    this.unFollowService = args.unFollowService;
+    this.resendTokenService = args.resendTokenService;
+  }
+
+  async createAccount(
+    name: string,
+    email: string,
+    passphrase: string,
+  ): Promise<
+    Result.Result<Error, z.infer<typeof CreateAccountResponseSchema>>
+  > {
+    const res = await this.registerAccountService.handle(
+      name as AccountName,
+      email,
+      '',
+      passphrase,
+      '',
+      'normal',
+    );
+
+    if (Result.isErr(res)) {
+      return Result.err(res[1]);
+    }
+
+    return Result.ok({
+      id: res[1].getID,
+      name: res[1].getName,
+      email: res[1].getMail,
+    });
+  }
+
+  async updateAccount(
+    name: string,
+    args: {
+      nickname?: string;
+      email?: string;
+      passphrase?: string;
+      bio: string;
+    },
+    etag: string,
+  ): Promise<
+    Result.Result<Error, z.infer<typeof UpdateAccountResponseSchema>>
+  > {
+    if (args.nickname) {
+      const res = await this.editAccountService.editNickname(
+        etag,
+        name as AccountName,
+        args.nickname,
+      );
+      if (Result.isErr(res)) {
+        return Result.err(res[1]);
```

Comment:
```suggestion
        return res;
```
取り出してから同じ `err` に詰め直していることになるので, これでもよいはずです. 他のコードも同様です.